c7n/resources/elasticsearch.py

```diff
@@ -426,6 +495,77 @@ class ElasticSearchMarkForOp(TagDelayedAction):
     """
 
 
+@ElasticSearchDomain.action_registry.register('remove-matched-source-ips')
+class RemoveMatchedSourceIps(BaseAction):
+    """Action to remove matched source ips from a Access Policy. This action
+    needs to be used in conjunction with the source-ip filter. It can be used
+    for removing non-approved IP addresses from the the access policy of a
+    ElasticSearch domain.
+
+    :example:
+    .. code-block:: yaml
+            policies:
+              - name: es-access-revoke
+                resource: elasticsearch
+                filters:
+                  - type: source-ip
+                    value_type: cidr
+                    op: not-in
+                    value_from:
+                       url: s3://my-bucket/allowed_cidrs.csv
+                actions:
+                  - type: remove-matched-source-ips
+    """
+
+    schema = type_schema('remove-matched-source-ips')
+    permissions = ('es:UpdateElasticsearchDomainConfig',)
+
```

Comment:
this probably should have a validate method to verify that the filter  is present (grep iter_filters for some examples)